CONTRIBUTING.md

```diff
@@ -1,22 +1,122 @@
-# Contributing to volplugin
+# Contributing to Volplugin
 
 Patches and feature additions are totally welcome to volplugin! For now, please
 see the [README](https://github.com/contiv/volplugin/blob/master/README.md) for
 build and test instructions.
 
-If you have a patch to contribute to volplugin, please follow these guidelines:
+There are other ways to contribute, not just code and feature additions
+- report issues, proposing documentation and design changes, submit bug
+fixes, propose design changes, discuss use cases or become a maintainer.
 
-* Sign your commits with `git commit -s`. Note that we live under an Apache
-  license.
-* Your comments should detail the directory they are affecting, as below. This
-  makes it much simpler than `git log --stat` to determine what areas of a
-  project a specific commit is referencing.
-  * Example: `librbd: changed a widget`
-* Please squash your commits per fix/feature, with a single commit message.
+## Submitting pull requests to change the documentation or the code
+
+Changes can be proposed by sending a pull request (PR). A maintainer
+will review the changes and provide feedback.
+
+The pull request will be merged into the master branch after discussion.
+
+Please make sure to run the tests and that the tests pass before
+submitting the PR. Please keep in mind that some changes might not be
+merged if the maintainers decide they can't be merged.
+
+Please squash your commits to one commit per fix or feature. The resulting
+commit should have a single meaningful message.
+
+## Commit message guidelines
+
+The commit should have a short summary of no more than 50 characters on
```

Comment:
let's leave the line limits out -- I don't really feel this is a necessary restriction, and I break this rule pretty much every commit.